### ISSUE TITLE
Use word-boundary matching for resolution label mismatch detection

### ISSUE THREAD

**Issue by kristofferR** [NewsGuyTor/IPTVChecker#39]
## Problem

`check_label_mismatch()` (line ~790) uses Python substring matching:
```python
if "4k" in channel_name_lower or "uhd" in channel_name_lower:
```

This can false-positive on channel names containing these as substrings (e.g., "E4Kids", "HD500", "CUHD"). The Rust version uses word-boundary regex matching to avoid this.

## Solution

Use `re.search()` with word boundaries:

```python
if re.search(r'\b4k\b', channel_name_lower) or re.search(r'\buhd\b', channel_name_lower):
    ...
elif re.search(r'\b1080p\b', channel_name_lower) or re.search(r'\bfhd\b', channel_name_lower):
    ...
elif re.search(r'\bhd\b', channel_name_lower):
    ...
```

## Impact
Reduces false-positive mismatch warnings for channels with resolution-like substrings in their names.
